subprojects/core/src/test/groovy/org/gradle/api/internal/classpath/DefaultModuleRegistryTest.groovy

```diff
@@ -83,6 +83,39 @@ class DefaultModuleRegistryTest extends Specification {
         module.runtimeClasspath.asFiles == [runtimeDep]
     }
 
+    def "locates module using manifest from runtime ClassLoader when run from IDEA with a project imported by IDEA"() {
+        given:
+        def classesDir = tmpDir.createDir("some-module/out/production/classes")
+        def staticResourcesDir = tmpDir.createDir("some-module/out/production/resources")
+        def ignoredDir = tmpDir.createDir("ignore-me-out/production/someModule")
+        def cl = classLoaderFor([ignoredDir, classesDir, resourcesDir, staticResourcesDir, runtimeDep])
+        def registry = new DefaultModuleRegistry(cl, ClassPath.EMPTY, null)
+
+        expect:
+        def module = registry.getModule("gradle-some-module")
+        module.implementationClasspath.asFiles == [classesDir, resourcesDir, staticResourcesDir]
+        module.runtimeClasspath.asFiles == [runtimeDep]
+    }
+
+    def "does not add wrong resources dirs to classpath of 'gradle-resources' module"() {
```

Comment:
Does this unit test make sense to check in? IIUC, you need to run this from IDEA to exercise one of the branches.

Reply:
I tried to figure out a test that reproduces the problem independent from wehre you run it. I was not successful.
I would like to leave this test in to document the issue. Even if you need to run it in the IDE. People who touch this this code again, will most likely run the test from the IDE.